src/test/java/core/basesyntax/service/impl/CsvParserTest.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.ProductTransaction;
+import core.basesyntax.enums.Operation;
+import core.basesyntax.service.TransactionParser;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CsvParserTest {
+    private static final List<String> VALID_TEST_DATA = Arrays.asList(
+            "operation,fruit,quantity",
+            "b,apple,10",
+            "s,banana,20",
+            "p,orange,5"
+    );
+    private static final List<String> EMPTY_TEST_DATA = Collections.emptyList();
+    private static final List<ProductTransaction> EXPECTED_TRANSACTIONS = Arrays.asList(
+            new ProductTransaction(Operation.BALANCE, "apple", 10),
+            new ProductTransaction(Operation.SUPPLY, "banana", 20),
+            new ProductTransaction(Operation.PURCHASE, "orange", 5)
+    );
+    private TransactionParser csvParser;
+
+    @BeforeEach
+    public void setUp() {
+        csvParser = new CsvParser();
+    }
```

Comment:
I don't think you need to create a new parser object before each test.

Reply:
Looks like it, changed it to :
```suggestion
    @BeforeAll
    static void beforeAll() {
        csvParser = new CsvParser();
    }
```